# Abundance Estimation of MAGs

Abundance estimation involves mapping sequencing reads to reference genomes or MAGs, with the number of mapped reads serving as a measure of genome abundance. To enable comparisons across samples, these counts are usually normalized by the total number of mapped reads per sample, providing relative abundance estimates. This approach facilitates the assessment of microbial diversity and community composition.

We will use Bowtie2 for read mapping and reference genome indexing, Samtools for BAM/SAM file processing, and msamtools to generate abundance tables.

### Tools overview

- **Bowtie2**: A fast and memory-efficient aligner for mapping sequencing reads to reference genomes or representative MAGs. It also performs indexing of the reference genomes, which is a required preprocessing step before alignment. Indexing optimizes the alignment process by converting the reference genome into a data structure that enables rapid and efficient read mapping.
- **Samtools**: A suite of tools for processing SAM/BAM files generated by Bowtie2. It is used for sorting, indexing, and filtering mapped reads to prepare data for quantification.
- **msamtools profile**: Quantifies genome abundance by analyzing read mapping coverage in a sorted BAM file. It calculates key metrics such as read depth and genome coverage, generating an abundance table that can be used for microbial community analysis.

### Setting Up Output Directories

Create a new folder named `09_abundance_estimation`. Inside this folder, create the following subdirectories:

üìÇ `09_abundance_estimation`/ <br>
‚îú‚îÄ‚îÄ üìÅ `bowtie_index_out`/ <br>
‚îú‚îÄ‚îÄ üìÅ `bowtie_mapping_out`/ <br>
‚îú‚îÄ‚îÄ üìÅ `msamtools_out`/ 


### Create the Bowtie2 index 

**Step 1: Concatenate Representative MAGs**

Before building the Bowtie2 index, we need a single FASTA file containing all representative MAGs. This requires merging individual MAG FASTA files into one. To generate this file, use the `concatenate_mags.sh` script located at `/hpcfs/home/cursos/bioinf-cabana/cabana_workshop/helper_scripts`. Copy the script to your `09_abundance_estimation` directory, update the `dereplicated_genomes` variable with the correct path, and execute it using `bash concatenate_mags.sh`.

This script will generate a file named `reference_genomes.fasta`, containing all merged MAGs.

**Step 2: Create and Execute the Bash Script for Bowtie2 Indexing**

To run Bowtie2 for index generation, create a Bash script named `run_bowtie2_index.sh`, copy the script below, and update the `reference_genomes` variable with the correct path to your previously generated FASTA file.

```
#!/bin/bash

#SBATCH -J bowtie_index
#SBATCH -D .
#SBATCH -e bowtie_index_%j.err
#SBATCH -o bowtie_index_%j.out
#SBATCH -n 8
#SBATCH --time=48:00:00	
#SBATCH --mem=3000

module load bowtie2/2.4.5 

reference_genomes="/path/to/reference_genomes.fasta"

bowtie2-build $reference_genomes bowtie_index_out/INDEX
```

After creating and saving the script, make it executable and submit to the cluster:

```
chmod +x run_bowtie2_index.sh
sbatch run_bowtie2_index.sh
```

### Output Decription

###  Create and Execute the Bash Script for Bowtie2 Mapping

### Output Decription

### Create the contig-to-bin mapping table for msamtools profile 

### Create and Execute the Bash Script to run msamtools

### Output Decription
