# Abundance Estimation of MAGs

Abundance estimation involves mapping sequencing reads to reference genomes or MAGs, with the number of mapped reads serving as a measure of genome abundance. To enable comparisons across samples, these counts are usually normalized by the total number of mapped reads per sample, providing relative abundance estimates. This approach facilitates the assessment of microbial diversity and community composition.

We will use Bowtie2 for read mapping and reference genome indexing, Samtools for BAM/SAM file processing, and msamtools to generate abundance tables.

### Tools overview

- **Bowtie2**: A fast and memory-efficient aligner for mapping sequencing reads to reference genomes or representative MAGs. It also performs indexing of the reference genomes, which is a required preprocessing step before alignment. Indexing optimizes the alignment process by converting the reference genome into a data structure that enables rapid and memory-efficient read mapping.
- **Samtools**: A suite of tools for processing SAM/BAM files generated by Bowtie2. It is used for sorting, indexing, and filtering mapped reads to prepare data for quantification.
- **msamtools profile**: Quantifies genome abundance by analyzing read mapping coverage in a sorted BAM file. It calculates key metrics such as read depth and genome coverage, generating an abundance table that can be used for microbial community analysis.

### Setting Up Output Directories

Create a new folder named `09_abundance_estimation`. Inside this folder, create the following subdirectories:

📂 `09_abundance_estimation`/ <br>
├── 📁 `bowtie_index_out`/ <br>
├── 📁 `bowtie_mapping_out`/ <br>
├── 📁 `msamtools_out`/ 


### Create the Bowtie2 index 

**Step 1: Concatenate Representative MAGs**

Before building the Bowtie2 index, all representative MAGs must be combined into a single FASTA file. Use the `concat_mags.sh script`, located at `/hpcfs/home/cursos/bioinf-cabana/cabana_workshop/helper_scripts`. Copy the script to your `09_abundance_estimation` directory, update the `dereplicated_genomes` variable with the correct path, and execute it using `bash concat_mags.sh`.

This will generate a file names `reference_genomes.fasta` in your current directory, containing all merged MAGs.

**Step 2: Create and Execute the Bash Script for Bowtie2 Indexing**

To run Bowtie2 for index generation, create a Bash script named `run_bowtie2_index.sh`, copy the script below, and update the `reference_genomes` variable with the correct path to your previously generated FASTA file. This script submits a SLURM job that builds the Bowtie2 index with the prefix INDEX inside the `bowtie_index_out directory`.

```
#!/bin/bash

#SBATCH -J bowtie_index
#SBATCH -D .
#SBATCH -e bowtie_index_%j.err
#SBATCH -o bowtie_index_%j.out
#SBATCH --cpus-per-task=8
#SBATCH --time=1:00:00	
#SBATCH --mem=3000

module load bowtie2/2.4.5 

reference_genomes="/path/to/reference_genomes.fasta"

bowtie2-build $reference_genomes bowtie_index_out/INDEX
```

After creating and saving the script, make it executable and submit to the cluster:

```
chmod +x run_bowtie2_index.sh
sbatch run_bowtie2_index.sh
```

### Output Decription

Once the indexing process is complete, the `bowtie_index_out` directory will contain the following files:

📂 `bowtie_index_out`/ <br>
│── 📄 `INDEX.1.bt2` <br>
│── 📄 `INDEX.2.bt2` <br>
│── 📄 `INDEX.3.bt2` <br>
│── 📄 `INDEX.4.bt2` <br>
│── 📄 `INDEX.rev.1.bt2` <br>
│── 📄 `INDEX.rev.2.bt2`

These files together form the FM-index (Ferragina-Manzini index), a compressed representation of the reference genomes that enables efficient sequence alignment.

- `INDEX.1.bt2`, `INDEX.2.bt2`, `INDEX.3.bt2`, `INDEX.4.bt2` – These are the forward strand index files generated by bowtie2-build. Bowtie2 uses these files to match sequencing reads to locations on the forward strand of the reference genomes.

- `INDEX.rev.1.bt2`, `INDEX.rev.2.bt2` – These are the reverse-complement index files, used for aligning reads that map to the reverse strand of the reference genomes.

###  Create and Execute the Bash Script for Bowtie2 Mapping

To map your sample reads to the set of representative genomes, create a Bash script named run_bowtie2_mapping.sh. Copy the script below and update the index and manifest variables with the correct paths to the Bowtie2 index you previously constructed and your cleaned reads manifest file, respectively.

```
#!/bin/bash

#SBATCH -J bowtie_mapping
#SBATCH -D .
#SBATCH -e bowtie_mapping_%j.err
#SBATCH -o bowtie_mapping_%j.out
#SBATCH --cpus-per-task=8
#SBATCH --time=1:00:00	
#SBATCH --mem=6000

module load bowtie2/2.4.5 samtools/1.16.1 
source /hpcfs/home/cursos/bioinf-cabana/conda/bin/activate
conda activate msamtools

manifest="/path/to/your/cleaned/reads/manifest.csv"
index="/path/to/bowtie_index_out/folder/"

# Loop through each line of the manifest file (skipping the header)
tail -n +2 "$manifest" | while IFS=',' read -r sample R1 R2; do
        
    # Bowtie2 mapping
    bowtie2 --threads 8 -x $index/INDEX -1 "$R1" -2 "$R2" -S bowtie_mapping_out/${sample}.sam

    # Convert SAM to BAM
    samtools view -F 4 -bS bowtie_mapping_out/${sample}.sam > bowtie_mapping_out/${sample}-RAW.bam

    # Using msamtools to filter aligments
    msamtools filter -b -l 80 -p 95 -z 80 bowtie_mapping_out/${sample}-RAW.bam > bowtie_mapping_out/${sample}.filtered.bam
    
    # Sort BAM file by name (needed for masamtools filter)
    samtools sort -n -@ 4 bowtie_mapping_out/${sample}.filtered.bam -o bowtie_mapping_out/${sample}.sorted.bam
   
    rm bowtie_mapping_out/${sample}.sam
    rm bowtie_mapping_out/${sample}-RAW.bam
    rm bowtie_mapping_out/${sample}.filtered.bam
    
done

```

After creating and saving the script, make it executable and submit to the cluster:

```
chmod +x run_bowtie2_mapping.sh
sbatch run_bowtie2_mapping.sh
```

This script submits a SLURM job to process each sample listed in the manifest file by:

1. Mapping paired-end reads to the set of representative MAGs (reference genomes) using bowtie2.
2. Converting the resulting SAM file to BAM format while filtering out reads that failed to map (-F 4).
3. Filtering alignments using msamtools, retaining only reads that:
  - Are at least 80 bp long (-l 80).
  - Have at least 95% identity with the MAG reference (-p 95).
  - Have at least 80% of the read aligned (-z 80).
3. Sorting the filtered BAM file by read name using samtools sort (required for msamtools profile).
4. Cleaning up intermediate files (.sam, -RAW.bam, .filtered.bam) to save storage space. 

### Output Decription

Once the mapping and filtering processes are complete, the `bowtie_mapping_out` directory will contain the following file:

📂 `bowtie_mapping_out`/ <br>
│── 📄 `MAG_ID.sorted.bam` 

This file stores high-confidence alignments to the reference genomes and serves as one of the input files for the final step: abundance estimation using msamtools profile.

### Create the contig-to-bin mapping table for msamtools profile 

### Create and Execute the Bash Script to run msamtools

### Output Decription
