# Assembling Metagenomic Reads with the metaWRAP Assembly Module

# Asembly module overwiew

The metaWRAP Assembly module enables users to assemble metagenomic reads using either metaSPAdes or MEGAHIT (default). During the process, short contigs (<1000 bp) are removed to improve assembly quality. Additionally, a assembly report is generated using QUAST, providing metrics for evaluating the results.

# Preparing the Manifest File

The reads stored in the `cleaned_reads` directory during the quality control step will serve as the input files for assembly. Therefore, a **new manifest** file should be created to reference these cleaned reads. Save the new manifest file in the `02_clean_reads` directory.

To construct this manifest file, follow the instructions here: [generate_manifest](
https://github.com/mariasotor/Cabana-Metagenomics-Workshop/blob/main/helper_scripts/generate_manifest.md)

### Setting Up Output Directories

Once the manifest file is ready, create a new folder named `03_assembly`. Inside this folder, create the following subdirectory:

ğŸ“‚ `03_assembly`/ <br>
â”œâ”€â”€ ğŸ“ `metawrap_assembly_out`/

### Create and Execute the Bash Script to Run the metaWRAP QC Module

To process the cleaned reads using the metaWRAP Asembly module, create a Bash script named `run_metawrap_assembly.sh`, copy the script below, and update the `manifest` variable with the correct path before running. This script submits a SLURM job to process each sample listed in the manifest file, assembling the reads using MEGAHIT and saving the results in the `metawrap_assembly_out` directory.

```
#!/bin/bash

#SBATCH -J metawrap_assembly
#SBATCH -D .
#SBATCH -e metawrap_assembly_%j.err
#SBATCH -o metawrap_assembly_%j.out
#SBATCH -n 8
#SBATCH --time=4:00:00	
#SBATCH --mem=6000	

source /hpcfs/home/cursos/bioinf-cabana/conda/bin/activate
conda activate metawrap-env

manifest="/path/to/your/clean/reads/manifest"

# Loop through each line of the manifest file (skipping the header)
tail -n +2 "$manifest" | while IFS=',' read -r sample R1 R2; do

metawrap assembly -1 $R1 -2 $R2 -m 6 -t 8 --megahit -o metawrap_assembly_out/$sample

done
```

After creating and saving the script, make it executable and submit it to the cluster:

```
chmod +x run_metawrap_assembly.sh
sbatch run_metawrap_assembly.sh
```

### Output Description

Once your samples have finished processing, you will see the following structure inside the `metawrap_assembly_out` folder:

ğŸ“‚ `metawrap_assembly_out`/ <br>
â”‚â”€â”€ ğŸ“‚ `sampleID`/ <br>
â”‚   â”œâ”€â”€ ğŸ“‚ `megahit`/ <br>
â”‚   â”œâ”€â”€ ğŸ“‚ `QUAST_out`/  <br>
â”‚   â”œâ”€â”€ ğŸ“„ `assembly_report.html` <br>
â”‚   â”œâ”€â”€ ğŸ“„ `final_assembly.fasta` 


- `megahit` â€“ Directory containing intermediate files generated by MEGAHIT during the assembly process.
- `QUAST_out` â€“ Directory containing the QUAST report, which provides assembly quality metrics such as N50, total length, and number of contigs.
- `assembly_report.html` â€“ This is an interactive HTML report generated by QUAST, summarizing the assembly quality statistics.
- `final_assembly.fasta` â€“ This is the final assembled contigs after filtering out short sequences (<1000 bp), ready for downstream analysis (binning).

