# Binning of Metagenomic Contigs and Bin Refinement with the metaWRAP Binning and Bin Refinement Modules

### Binning Module Overview
The metaWRAP Binning module simplifies metagenomic binning by integrating three binning tools: MaxBin2, metaBAT2, and CONCOCT. It performs the following steps:

- Indexes the metagenomic assembly for paired-end reads (sample) alignment.
- Runs the specified binning tools.
- Assesses bin quality using CheckM.
- Generates a summary report.

### Bin Refinement Module Overview
The metaWRAP Bin Refinement module improves binning results by merging and optimizing bin sets from different binning tools. It performs the following steps:

- Iteratively compares and merges bin sets.
- Identifies bins with at least 80% genome overlap and selects the better bin using the formula: Score = Completion - (5 Ã— Contamination).
- Removes duplicate contigs, ensuring each contig remains only in the highest-quality bin.
- Evaluates refined bins with CheckM and generates a summary report.

### Setting Up Output Directories

Create a new folder named `04_binning_and_refinement`. Inside this folder, create two subdirectorie:

ğŸ“‚ `04_binning_and_refinement`/ <br>
â”œâ”€â”€ ğŸ“ `binning`/ <br>
â”œâ”€â”€ ğŸ“ `refinement`/

### Create and Execute the Bash Script to Run the metaWRAP Binning and Bin Refinement Modules

To perform the binning of contigs and subsequent bin refinemnet, create a Bash script named `run_metawrap_binning_and_refinement.sh`, copy the script below, and update the `manifest` and `assembly_folder` variables with the correct path before running. 

This script submits a SLURM job to perform  two processes:
1. Binning: Runs MetaBAT2 and MaxBin2 on the assembled contigs and the cleaned reads from each sample listed in the manifest file. The output is saved in the `binning` directory.
2. Bin Refinement: Refines the bins generated by MetaBAT2 and MaxBin2, keeping only bins of medium to high quality (â‰¥50% completeness and â‰¤5% contamination). The output is saved in the `refinement` directory.

```
#!/bin/bash

#SBATCH -J metawrap_binning_and_refinement
#SBATCH -D .
#SBATCH -e metawrap_binning_and_refinement_%j.err
#SBATCH -o metawrap_binning_and_refinement_%j.out
#SBATCH -n 8
#SBATCH --time=6:00:00	
#SBATCH --mem=16000	

source /hpcfs/home/cursos/bioinf-cabana/conda/bin/activate
conda activate metawrap-env

manifest="path/to/your/clean/reads/manifest"
assembly_folder="path/to/metawrap_assembly_out"

# Loop through each line of the manifest file (skipping the header)
tail -n +2 "$manifest" | while IFS=',' read -r sample R1 R2; do

metawrap binning -o binning/$sample -t 8 -m 8 -a $assembly_folder/$sample/final_assembly.fasta --metabat2 --maxbin2 $R1 $R2
metawrap bin_refinement -o refinement/$sample -t 8 -m 16 -A binning/$sample/metabat2_bins/ -B binning/$sample/maxbin2_bins/ -c 50 -x 5 --quick

done
```

After creating and saving the script, make it executable and submit it to the cluster:

```
chmod +x run_metawrap_binning_and_refinement.sh
sbatch run_metawrap_binning_and_refinement.sh
```

### Output Description

Once the binning and bin refinement procesess have finished, you will see the following structure inside the `04_binning_and_refinement` directory:

ğŸ“‚ `04_binning_and_refinement`/ <br>
â”‚â”€â”€ ğŸ“‚ `binning`/ <br>
â”‚   â”œâ”€â”€ ğŸ“‚ `SampleID`/ <br>
â”‚     â”œâ”€â”€ ğŸ“‚ `maxbin2_bins`/ <br>
â”‚     â”œâ”€â”€ ğŸ“‚ `metabat2_bins`/  <br>
â”‚     â”œâ”€â”€ ğŸ“‚ `work_files` <br>
â”‚ <br>
â”‚â”€â”€ ğŸ“‚ `refinement`/ <br>
â”‚   â”œâ”€â”€ ğŸ“‚ `SampleID`/ <br>
â”‚     â”œâ”€â”€ ğŸ“‚ `figures`/ <br>
â”‚     â”œâ”€â”€ ğŸ“‚ `maxbin2_bins` <br>
â”‚     â”œâ”€â”€ ğŸ“‚ `metabat2_bins`/ <br>
â”‚     â”œâ”€â”€ ğŸ“‚ `metawrap_50_5_bins`/ <br>
â”‚     â”œâ”€â”€ ğŸ“‚ `work_files`/ <br>
â”‚     â”œâ”€â”€ ğŸ“„ `maxbin2_bins.contigs` <br>
â”‚     â”œâ”€â”€ ğŸ“„ `maxbin2_bins.stats` <br>
â”‚     â”œâ”€â”€ ğŸ“„ `metabat2_bins.contigs` <br>
â”‚     â”œâ”€â”€ ğŸ“„ `metabat2_bins.stats` <br>
â”‚     â”œâ”€â”€ ğŸ“„ `metawrap_50_5_bins.contigs` <br> 
â”‚     â”œâ”€â”€ ğŸ“„ `metawrap_50_5_binsstats` 


- `maxbin2_bins` and `metabat2_bins` â€“ Genome bins (FASTA files) generated using MaxBin2 and MetaBAT2, respectively. 
- `binning/SampleID/work_files/` â€“ Intermediate files generated during binning, such as contig abundance calculations and summary report files.
- `figures` â€“ Plots for bin quality assessment before and after refinement.
- `metawrap_50_5_bins` â€“ The final refined bins with â‰¥50% completeness and â‰¤5% contamination, ready for downstream analysis.
- `refinement/SampleID/work_files` â€“ Intermediate files created during the refinement process.
- `maxbin2_bins.contigs`, `metabat2_bins.contigs`, `metawrap_50_5_bins.contigs` â€“ Files specifying which contig belongs to which bin.
- `maxbin2_bins.stats`, `metabat2_bins.stats`, `metawrap_50_5_bins.stats` â€“ Summary statistics of bins from each binner and after refinement (e.g., completeness, contamination, GC content, size).

To facilitate downstream analysis, the sey of bins in the `metawrap_50_5_bins` directory should be renames as sampleID_bin.1.fa, sampleID_bin.2.fa...

To rename the files, use the `rename_bins.sh` script located at located at /hpcfs/home/cursos/bioinf-cabana/cabana_workshop/helper_scripts. Copy this script to your `04_binning_and_refinement` directory, open it, and update the `parent_folder` variable with the correct path. Then, run the script using `bash rename_bins.sh`.


