# Bioinformatics Tools Used in the Workshop

For detailed information about each bioinformatics tool used in this workshop, please refer to the corresponding documentation links below:

**Reads quality control, assembly and binning**

  - *metaWRAP*
    - https://github.com/bxlab/metaWRAP
    - https://github.com/bxlab/metaWRAP/blob/master/Module_descriptions.md
 
**MAGs quality control**

  - *MDMcleaner*
    - https://github.com/KIT-IBG-5/mdmcleaner

 - *GUNC*
     - https://grp-bork.embl-community.io/gunc/
  
 - *CheckM2*
    - https://github.com/chklovski/CheckM2

**MAGs dereplication**

  - *dRep*
      - https://drep.readthedocs.io/en/latest/module_descriptions.html
    
**MAGs abundance estimation**

  - *Bowtie 2*
      - https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#building-from-source

  - *Samtools*
     - https://www.htslib.org/doc/samtools.html
  
  - *msamtools*
      - https://github.com/arumugamlab/msamtools

**MAGs taxonomy classificatioon**

  - *GTDBtk*
      - https://ecogenomics.github.io/GTDBTk/
      - https://ecogenomics.github.io/GTDBTk/commands/classify_wf.html

**MAGs functional annotation**

  - *Bakta*
      - https://github.com/oschwengers/bakta

  - *AMRFinderPlus*
      - https://github.com/ncbi/amr/wiki
      - https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#usage

