# Comparison Between Two Metagenomic Workflows for MAG Reconstruction and Quality Control

This document presents a comparison between the metagenomic analysis workflow used in this course and the one applied in the CABANAnet project *"Exploring the Human Gut Microbiome Diversity in Latin America: Focus on Populations at Nutritional Risk"*.

While the two workflows are highly similar, the version used in this course has been modified to reduce computing resource usage (RAM, number of threads, and processing time), allowing all analyses to be completed efficiently within the available timeframe and cluster resources.

The two metagenomic workflows, along with estimated resource usage for each tool, are presented in the diagrams below. Both workflows were implemented on the same metagenomic sample containing 20 million reads.

**Original Workflow**
![metagenomics_workflow_comp_orig](https://github.com/user-attachments/assets/5049596b-58ae-4103-b1be-d08bf8fed9c2)

**Modified Workflow**
![metagenomics_workflow_comp](https://github.com/user-attachments/assets/8ba1f0f2-2c66-4c44-a0e7-acc3837487ed)
